In [54]:
#import modules引入模組
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#讀資料
#import dataset
#read_csv：local or internet url OK
df = pd.read_csv("https://raw.githubusercontent.com/ryanchung403/dataset/main/Housing_Dataset_Sample.csv")

df.head(n=6)
df.describe().T



X = df.iloc[:,:5]
y = df['Price']
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.38, random_state=45)

from sklearn.linear_model import LinearRegression
reg = LinearRegression()
reg.fit(X_train,y_train)
prediction =reg.predict(X_test)






In [ ]:
sns.distplot(df['Price'])
sns.displot(df['Price'])
sns.jointplot(x=df['Avg. Area Income'],y=df['Price'])
sns.pairplot(df)

In [55]:

from sklearn.metrics import r2_score
r2_score(y_test, prediction)

0.9222818906182605

In [ ]:
plt.scatter(y_test, prediction, color='green', alpha=0.3)

測試結果：
test_size=0.1, random_state=48 ,0.9156535

test_size=0.3, random_state=46 ,0.9216613392669022

test_size=0.2, random_state=64 ,0.9162279902509566

test_size=0.3, random_state=64 ,0.9154468136730208

test_size=0.4, random_state=64 ,0.9177585913164295

test_size=0.5, random_state=44 ,0.9186657748247419

test_size=0.5, random_state=54 ,0.9198740696104669

test_size=0.5, random_state=64 ,0.915323879907813

test_size=0.7, random_state=40 ,0.9179534902072216

test_size=0.4, random_state=45 ,0.9217813719018431

test_size=0.5, random_state=43 ,0.9205983001061324

test_size=0.45, random_state=44 ,0.9178003563563922

test_size=0.01, random_state=44 ,0.9027641399956966

test_size=0.9, random_state=44 ,0.9182455890754818

test_size=0.9, random_state=60 ,0.9167613306547876

test_size=0.9, random_state=5 ,0.9169041203891759

test_size=0.9, random_state=99 ,0.9153325253185588

test_size=0.35, random_state=45 ,0.9220341809599634

test_size=0.36, random_state=45 ,0.922356461559385

test_size=0.38, random_state=45 ,0.9222818906182605

目前測試結果最高為0.9222818906
推估可得最大值之test_size為0.36,random_state為45之組合
testsize為最靠近1之大值和最靠近0之小值並無得到較大結果趴數